unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/Environment.cs

```diff
@@ -19,18 +20,34 @@ public class Environment
         private Environment()
         {
             messagingControllersManager = new MessagingControllersManager();
+            pointerEventsController = new PointerEventsController();
+            memoryManager = new MemoryManager();
         }
 
-        public void Initialize(IMessageProcessHandler messageHandler)
+        public void Initialize(IMessageProcessHandler messageHandler, bool isTesting = false)
         {
             messagingControllersManager.Initialize(messageHandler);
+            pointerEventsController.Initialize(isTesting);
+            memoryManager.Initialize();
         }
 
-        public void Restart(IMessageProcessHandler messageHandler)
+        public void Cleanup()
         {
             messagingControllersManager.Cleanup();
+            pointerEventsController.Cleanup();
+            memoryManager.CleanupPoolsIfNeeded(true);
+        }
 
-            this.Initialize(messageHandler);
+        public void Restart(IMessageProcessHandler messageHandler, bool isTesting = false)
+        {
+            this.Cleanup();
+            this.Initialize(messageHandler, isTesting);
+        }
+
+
+        public void InitializeForTesting()
```

Comment:
As this is used for testing, this code would be better encapsulated if we put this method on the testing suite.

Reply:
Agree, as I didn't know where to put it, I leave it here for now. Where should it be moved?

Reply:
My best bet is to move this to the test class or `TestsBase`.

Reply:
Just moved it